### PR TITLE
Remove "uninverted" and "binary" fielddata support for numeric and boolean fields.

### PR DESCRIPTION
**Pull request by jpountz**
Numeric and boolean fields have doc values enabled by default as of
elasticsearch 2.0. This commit removes support for uninverted/in-memory
fielddata, as well as numeric fields encoded in binary doc values which was
the way that elasticsearch stored doc values in a Lucene index before the
1.4 release.

As a consequence, you will only be able to sort and aggregate on numeric and
boolean fields in Elasticsearch 3.0 if doc values have not been switched off.

Relates #14113

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/index/fielddata/IndexFieldDataService.java [hunk 1]
```diff
@@ -64,9 +61,17 @@
     public static final String FIELDDATA_CACHE_KEY = "index.fielddata.cache";
     public static final String FIELDDATA_CACHE_VALUE_NODE = "node";
 
+    private static final IndexFieldData.Builder MISSING_DOC_VALUES_BUILDER = new IndexFieldData.Builder() {
+        @Override
+        public IndexFieldData<?> build(Index index, Settings indexSettings, MappedFieldType fieldType, IndexFieldDataCache cache, CircuitBreakerService breakerService, MapperService mapperService) {
+            throw new IllegalStateException("Fielddata can't be loaded on field [" + fieldType.names().fullName()
```
Comment (nik9000) [inline core/src/main/java/org/elasticsearch/index/fielddata/IndexFieldDataService.java]: I'm not sure what this message means - maybe better to say that fielddata is unsupported for these types and to use docvalues instead?

File: core/src/test/java/org/elasticsearch/index/fielddata/IndexFielddataServiceTests.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            MappedFieldType ft = new LongFieldMapper.LongFieldType();
+            ft.setNames(new Names("some_long"));
+            ft.setHasDocValues(true);
+            ifds.getForField(ft); // no exception
+            ft.setHasDocValues(false);
+            try {
+                ifds.getForField(ft);
+                fail();
+            } catch (IllegalStateException e) {
+                assertTrue(e.getMessage().contains("doc values"));
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/index/fielddata/IndexFielddataServiceTests.java]: `assertThat(e.getMessage(), containsString("doc values"))` provides better error messages on failure.

**Comment by jpountz** [comment]
@nik9000 I pushed a new commit.

File: core/src/test/java/org/elasticsearch/index/fielddata/IndexFieldDataServiceTests.java [hunk 3]
```diff
@@ -238,4 +188,67 @@ public void onRemoval(ShardId shardId, MappedFieldType.Names fieldNames, FieldDa
         }
     }
 
+
+    public void testRequireDocValuesOnLongs() {
+        ThreadPool threadPool = new ThreadPool("random_threadpool_name");
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/index/fielddata/IndexFieldDataServiceTests.java]: These seem copy-and-paste-ish. Can you make a method that does the assertion and just call it for all the types? Or do it in a for loop or something?

**Comment by nik9000** [comment]
Left a minor comment - it looks pretty good to me otherwise. Its nice to see this code go.

**Pull request merged**